card/mod/core/set/all/haml.rb

```diff
@@ -1,7 +1,82 @@
 format do
-  def render_haml locals={}, template=nil, a_binding=nil, &block
-    template ||= yield
+  # Renders haml templates. The haml template can be passed as string or
+  # block or a symbol that refers to a view template.
+  # @param  template_or_locals [Hash, String, Symbol]
+  #   If a symbol is given then a template is expected in the corresponding view
+  #   directory. Note that {view_template_path} needs to be overridden in
+  #   that case to get the right path to the template.
+  # @return [String] rendered haml as HTML
+  # @example render a view template
+  #   # view/type/basic/my_template.haml:
+  #   %p
+  #     Hi
+  #     = name
+  #
+  #   # set/type/basic.rb:
+  #   view :my_view do
+  #     render_haml :my_template, name: "Joe:  # => "<p>Hi Joe<p/>"
+  #   end
+  # @example use a block to pass haml
+  #   render_haml name: "Joe" do
+  #     <<-HAML.strip_heredoc
+  #       %p
+  #         Hi
+  #         = name
+  #     HAML
+  #   # => <p>Hi Joe</p>
+  # @example create a slot in haml code
+  #   - haml_wrap do
+  #     %p
+  #       some haml
+  def render_haml template_or_locals={}, locals_or_binding=nil, a_binding=nil
```

Comment:
As much as we overuse "args", I'm wondering if it wouldn't actually be a little clearer in this case.  Something like:

```
def render_haml *args
   return render_haml_template(*args) if args.first.is_a? Symbol
   haml = block_given? ? yield : args.shift
   haml_to_html haml, *args
end
```